packages/flutter_tools/test/commands.shard/hermetic/pub_get_test.dart

```diff
@@ -123,3 +150,30 @@ class FakePub extends Fake implements Pub {
       ..writeAsStringSync('{"configVersion":2,"packages":[]}');
   }
 }
+
+class FakePubTargetDirectory extends Fake implements Pub {
```

Comment:
What about adding an optional parameter to the `FakePub` constructor for `currentDirectory` that defaults to `fileSystem.currentDirectory`, rather than creating a whole new class?

Reply:
> Oh wait, actually, doesn't this already get piped through as the `directory` parameter to `.get()`?

I do not follow. Do you mean that fileSystem.currentDirectory... is not needed? This syncs are needed for the currentDirectory to be a "valid flutter project"

Reply:
Ohh I see what you mean yes I can just use the directory instead which is already in the function. Good call